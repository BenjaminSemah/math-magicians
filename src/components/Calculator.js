import React, { useState } from 'react';
import calculate from '../logic/calculate';

const Calculator = () => {
  const [state, setState] = useState({
    total: '0',
    next: '',
    operation: '',
  });

  const clickHandler = (event) => {
    const buttonName = event.target.id;
    const { total, next, operation } = calculate(state, buttonName);
    if (next === null && total === null) {
      setState({ next, total: '0', operation });
    } else {
      setState({ next, total, operation });
    }
  };

  return (
    <div className="inside--calculator">
      <p>Let&apos;s do some math!</p>
      <div className="main--container">
        <div className="display--panel">{state.next || state.total || '0'}</div>
        <button onClick={clickHandler} id="AC" type="button" className="ac--btn grey">AC</button>
        <button onClick={clickHandler} id="+/-" type="button" className="add-sub grey">+/-</button>
        <button onClick={clickHandler} id="%" type="button" className="modular grey">%</button>
        <button onClick={clickHandler} id="รท" type="button" className="divi calc">รท</button>
        <button onClick={clickHandler} id="7" type="button" className="num7 grey">7</button>
        <button onClick={clickHandler} id="8" type="button" className="num8 grey">8</button>
        <button onClick={clickHandler} id="9" type="button" className="num9 grey">9</button>
        <button onClick={clickHandler} id="x" type="button" className="times calc">x</button>
        <button onClick={clickHandler} id="4" type="button" className="num4 grey">4</button>
        <button onClick={clickHandler} id="5" type="button" className="num5 grey">5</button>
        <button onClick={clickHandler} id="6" type="button" className="num6 grey">6</button>
        <button onClick={clickHandler} id="-" type="button" className="minus calc">-</button>
        <button onClick={clickHandler} id="1" type="button" className="num1 grey">1</button>
        <button onClick={clickHandler} id="2" type="button" className="num2 grey">2</button>
        <button onClick={clickHandler} id="3" type="button" className="num3 grey">3</button>
        <button onClick={clickHandler} id="+" type="button" className="plus calc">+</button>
        <button onClick={clickHandler} id="0" type="button" className="num0 grey">0</button>
        <button onClick={clickHandler} id="." type="button" className="point grey">.</button>
        <button onClick={clickHandler} id="=" type="button" className="equal calc">=</button>
      </div>
    </div>
  );
};

export default Calculator;
