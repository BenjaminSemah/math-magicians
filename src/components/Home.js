import React from 'react';

export default function Home() {
  return (
    <div className="inside--home">
      <h1>Wecome! Cheers to having fun with Maths!</h1>
      <p>
        Having some math knowledge is always useful.
        If you think about it, most of us use it on a
        daily basis — whether we are measuring something,
        sorting, noticing patterns, making comparisons or
        recognizing shapes. Of course, we do not necessarily
        use complex formulas, but we do need some basic math
        skills for all these activities.
      </p>
      <p>
        When it comes to children, it is crucial to teach them
        about the importance of problem solving. We all know
        that for some kids there are few words scarier than
        “math”. But we can influence their thinking by
        incorporating math in enjoyable activities and
        by having a positive attitude towards it.
      </p>
      <p>Source: Goalcast.com</p>
    </div>
  );
}
