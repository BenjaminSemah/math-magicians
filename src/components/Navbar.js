import React from 'react';
import { Outlet, Link } from 'react-router-dom';

export default function Navbar() {
  return (
    <div className="topbar">
      <div className="inside--nav">
        <h2>Math Magicians</h2>
        <nav className="nav--links">
          <Link to="/home">Home </Link>
          <Link to="/calculator">Calculator </Link>
          <Link to="/quote">Quote</Link>
        </nav>
      </div>
      <Outlet />
    </div>
  );
}
