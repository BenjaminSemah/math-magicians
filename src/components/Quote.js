import React from 'react';

export default function Quote() {
  return (
    <div className="inside--quote">
      <q>
        A man is like a fraction whose
        numerator is what he is and whose
        denominator is what he thinks of
        himself. The larger the denominator,
        the smaller the fraction.
      </q>
      <p>- Leo Tolstoy</p>
    </div>
  );
}
