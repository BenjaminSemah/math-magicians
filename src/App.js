import React from 'react';
import { BrowserRouter as Router, Route, Routes } from 'react-router-dom';
import Navbar from './components/Navbar';
import Home from './components/Home';
import Calculator from './components/Calculator';
import Quote from './components/Quote';

const App = () => (
  <Router>
    <Navbar />
    <Routes>
      <Route path="/" exact element={<Home />} />
      <Route path="/calculator" element={<Calculator />} />
      <Route path="/quote" element={<Quote />} />
    </Routes>
  </Router>
);

export default App;
