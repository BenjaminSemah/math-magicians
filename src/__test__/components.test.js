import React from 'react';
import renderer from 'react-test-renderer';
import App from '../App';
import Home from '../components/Home';
import Quote from '../components/Quote';
import Calculator from '../components/Calculator';
import Navbar from '../components/Navbar';

it('App renders correctly', () => {
  const tree = renderer.create(<App />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Home renders correctly', () => {
  const tree = renderer.create(<Home />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Calculator renders correctly', () => {
  const tree = renderer.create(<Calculator />).toJSON();
  expect(tree).toMatchSnapshot();
});

it('Quote renders correctly', () => {
  const tree = renderer.create(<Quote />).toJSON();
  expect(tree).toMatchSnapshot();
});
