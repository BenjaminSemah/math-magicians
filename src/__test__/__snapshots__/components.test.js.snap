// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders correctly 1`] = `
Array [
  <div
    className="topbar"
  >
    <div
      className="inside--nav"
    >
      <h2>
        Math Magicians
      </h2>
      <nav
        className="nav--links"
      >
        <a
          className="nav--item bod"
          href="/"
          onClick={[Function]}
        >
          Home 
        </a>
        <a
          className="nav--item bod"
          href="/calculator"
          onClick={[Function]}
        >
          Calculator 
        </a>
        <a
          className="nav--item"
          href="/quote"
          onClick={[Function]}
        >
          Quote
        </a>
      </nav>
    </div>
  </div>,
  <div
    className="inside--home"
  >
    <h1>
      Wecome! Cheers to having fun with Maths!
    </h1>
    <p>
      Having some math knowledge is always useful. If you think about it, most of us use it on a daily basis — whether we are measuring something, sorting, noticing patterns, making comparisons or recognizing shapes. Of course, we do not necessarily use complex formulas, but we do need some basic math skills for all these activities.
    </p>
    <p>
      When it comes to children, it is crucial to teach them about the importance of problem solving. We all know that for some kids there are few words scarier than “math”. But we can influence their thinking by incorporating math in enjoyable activities and by having a positive attitude towards it.
    </p>
    <p>
      Source: Goalcast.com
    </p>
  </div>,
]
`;

exports[`Calculator renders correctly 1`] = `
<div
  className="inside--calculator"
>
  <p>
    Let's do some math!
  </p>
  <div
    className="main--container"
  >
    <div
      className="display--panel"
    >
      0
    </div>
    <button
      className="ac--btn grey"
      id="AC"
      onClick={[Function]}
      type="button"
    >
      AC
    </button>
    <button
      className="add-sub grey"
      id="+/-"
      onClick={[Function]}
      type="button"
    >
      +/-
    </button>
    <button
      className="modular grey"
      id="%"
      onClick={[Function]}
      type="button"
    >
      %
    </button>
    <button
      className="divi calc"
      id="÷"
      onClick={[Function]}
      type="button"
    >
      ÷
    </button>
    <button
      className="num7 grey"
      id="7"
      onClick={[Function]}
      type="button"
    >
      7
    </button>
    <button
      className="num8 grey"
      id="8"
      onClick={[Function]}
      type="button"
    >
      8
    </button>
    <button
      className="num9 grey"
      id="9"
      onClick={[Function]}
      type="button"
    >
      9
    </button>
    <button
      className="times calc"
      id="x"
      onClick={[Function]}
      type="button"
    >
      x
    </button>
    <button
      className="num4 grey"
      id="4"
      onClick={[Function]}
      type="button"
    >
      4
    </button>
    <button
      className="num5 grey"
      id="5"
      onClick={[Function]}
      type="button"
    >
      5
    </button>
    <button
      className="num6 grey"
      id="6"
      onClick={[Function]}
      type="button"
    >
      6
    </button>
    <button
      className="minus calc"
      id="-"
      onClick={[Function]}
      type="button"
    >
      -
    </button>
    <button
      className="num1 grey"
      id="1"
      onClick={[Function]}
      type="button"
    >
      1
    </button>
    <button
      className="num2 grey"
      id="2"
      onClick={[Function]}
      type="button"
    >
      2
    </button>
    <button
      className="num3 grey"
      id="3"
      onClick={[Function]}
      type="button"
    >
      3
    </button>
    <button
      className="plus calc"
      id="+"
      onClick={[Function]}
      type="button"
    >
      +
    </button>
    <button
      className="num0 grey"
      id="0"
      onClick={[Function]}
      type="button"
    >
      0
    </button>
    <button
      className="point grey"
      id="."
      onClick={[Function]}
      type="button"
    >
      .
    </button>
    <button
      className="equal calc"
      id="="
      onClick={[Function]}
      type="button"
    >
      =
    </button>
  </div>
</div>
`;

exports[`Home renders correctly 1`] = `
<div
  className="inside--home"
>
  <h1>
    Wecome! Cheers to having fun with Maths!
  </h1>
  <p>
    Having some math knowledge is always useful. If you think about it, most of us use it on a daily basis — whether we are measuring something, sorting, noticing patterns, making comparisons or recognizing shapes. Of course, we do not necessarily use complex formulas, but we do need some basic math skills for all these activities.
  </p>
  <p>
    When it comes to children, it is crucial to teach them about the importance of problem solving. We all know that for some kids there are few words scarier than “math”. But we can influence their thinking by incorporating math in enjoyable activities and by having a positive attitude towards it.
  </p>
  <p>
    Source: Goalcast.com
  </p>
</div>
`;

exports[`Quote renders correctly 1`] = `
<div
  className="inside--quote"
>
  <q>
    A man is like a fraction whose numerator is what he is and whose denominator is what he thinks of himself. The larger the denominator, the smaller the fraction.
  </q>
  <p>
    - Leo Tolstoy
  </p>
</div>
`;
